aws_lambda_builders/workflows/nodejs_npm/npm.py

```diff
@@ -0,0 +1,53 @@
+"""
+Wrapper around calling npm through a subprocess.
+"""
+
+import logging
+
+from .utils import OSUtils
+
+LOG = logging.getLogger(__name__)
+
+
+class NpmError(Exception):
+    pass
+
+
+class NpmExecutionError(NpmError):
+    def __init__(self, err):
+        super(NpmExecutionError, self).__init__(
+            'NPM failed: %s' % err)
+
+
+class SubprocessNpm(object):
+
+    def __init__(self, osutils=None, npm_exe=None):
+        if osutils is None:
+            osutils = OSUtils()
+        self.osutils = osutils
+
+        if npm_exe is None:
+            npm_exe = 'npm'
```

Comment:
In the future if there are multiple paths of npm on your system, how would you imagine passing them in? or maybe thats not a big concern.

I started a PR here: https://github.com/awslabs/aws-lambda-builders/pull/35 that would help with that. But I'm going to take a step back and maybe write a mini design doc to get started on it. I was curious to know if you had any comments on that?